reports/report.md

```diff
@@ -4670,12 +4670,14 @@ Contract contains comments with TODOS
 
 - Found in src/ContractWithTodo.sol [Line: 4](../tests/contract-playground/src/ContractWithTodo.sol#L4)
 
+	A TODO comment was found on line 8
 	```solidity
 	contract TodoList {
 	```
 
 - Found in src/Counter.sol [Line: 4](../tests/contract-playground/src/Counter.sol#L4)
 
+	A TODO comment was found on line 15
```

Comment:
@TilakMaddy , this still doesn't feel quite right to me.

1. The `ContractDefinition` is captured in this detector, which is on line 4
2. This "hint" suggests line 15 (confusing for a developer reading both lines 4 and 15)
3. The Counter contract actually has 3 instances of `// TODO`, line 15, 20 and 22. This is not captured in hints.

IMO, simply altering this detector to detect the actual lines that `// TODO` is on would be preferable over adding this "hints" feature in its current form.

Reply:
> IMO, simply altering this detector to detect the actual lines that // TODO is on would be preferable over adding this "hints" feature in its current form.

For this case, yes I agree ^^ 

But I think I'll make a better case with some other detector as example. 
For instance:
When we capture the for loop, it has a problematic function call somewhere deep in the downward callgraph we can specify that as a `hint`

I think without, that there is too much work user has to do find exactly why the for loop was captured.


Coming back to this case, only line 15 captured, but after that correction is made, then line 20 will be shown.
So it's incremental. Just like clippy. Sometimes you have correct what it shows you first, to be able to see more things that go wrong.


Shall I undo the changes for todo detector ?

What do you think about the general structure though for `hints` (irrespective of the detector example) ? @alexroan

Reply:
- Given that it highlights the captured instances, how would this feature be applied in the VSCode extension?
- Why are "hints" required instead of simply making the detect function more precise? And can you show me an example of a detector where it is?

Reply:
*In VSCode, Hints would be appended to the general description, but only for that particular instance.
(Read above for more explanation)

* Okay, I will add a better example to this PR :)

Reply:
✅
DONE